src/data_type.hpp

```diff
@@ -452,18 +435,37 @@ class UserType : public DataType {
   CaseInsensitiveHashTable<Field> fields_;
 };
 
-class NativeDataTypes {
+class ValueTypes {
 public:
-  void init_class_names();
-  const DataType::ConstPtr& by_class_name(const std::string& name) const;
+  ValueTypes();
+
+  static CassValueType by_class(StringRef name);
+  static CassValueType by_cql(StringRef name);
+
+private:
+  typedef SmallDenseHashMap<StringRef, CassValueType,
+                            CASS_VALUE_TYPE_LAST_ENTRY, // Max size
+                            StringRefIHash,
+                            StringRefIEquals> HashMap;
+
+  static HashMap value_types_by_class_;
+  static HashMap value_types_by_cql_;
+};
+
+class SimpleDataTypeCache {
+public:
+  const DataType::ConstPtr& by_class(StringRef name) {
```

Comment:
Do we need reference counting for this? What if it's a singleton initialized at the beginning of time and consumers get the elements they need? The consumers don't even try to delete what they've got, and when the application terminates, the singleton will be destroyed then. by_class and similar methods could return a `const DataType&` or `const DataType*`; the former would make it less likely that a consumer would try to destroy the DataType.